pkg/ovs/openflow/interfaces.go

```diff
@@ -198,6 +199,11 @@ type FlowBuilder interface {
 	MatchTCPDstPort(port uint16) FlowBuilder
 	MatchUDPDstPort(port uint16) FlowBuilder
 	MatchSCTPDstPort(port uint16) FlowBuilder
+	MatchCTSrcIP(ip net.IP) FlowBuilder
```

Comment:
Will it be useful to have a match for IPNet too as it supports masking: MatchCTSrcIP? This may help with network policies with CIDRs I guess.

Reply:
Since this field are used to record the original IP in the connection, OVS doesn't support IPNet. If configuring the OpenFlow entry with IPNet, an error like "actions are invalid with specified match (OFPBAC_MATCH_INCONSISTENT)" might be returned.

Reply:
```       Connection Tracking Original Direction IPv4 Source Address Field

       Name:            ct_nw_src
       Width:           32 bits
       Format:          IPv4
       Masking:         arbitrary bitwise masks
       Prerequisites:   CT
       Access:          read-only
       OpenFlow 1.0:    not supported
       OpenFlow 1.1:    not supported
       OXM:             none
       NXM:             NXM_NX_CT_NW_SRC (120) since Open vSwitch 2.8

       Matches IPv4 conntrack original direction tuple source address. See the
       paragraphs above for general description to the conntrack original  di‐
       rection tuple. Introduced in Open vSwitch 2.8.
```

@wenyingd I am a bit confused by your answer to @srikartati. If this is the field described above, according to the doc it should support arbitrary masks. Are we talking about a different field here?

Reply:
> Since this field is used to record the original IP in the connection, OVS doesn't support IPNet. If configuring the OpenFlow entry with IPNet, an error like "actions are invalid with specified match (OFPBAC_MATCH_INCONSISTENT)" might be returned.

I have same question as @antoninbas . I looked at the same field yesterday and saw the bitmask to be supported.
https://manpages.debian.org/unstable/openvswitch-common/ovs-fields.7.en.html

Reply:
I tried following ovs-ofctl command on OVS, and got these errors:

Reply:
it could be a documentation issue, but I feel like the error message would be different, as it is it doesn't really make sense to me. Could you reach out to the OVS folks?

Reply:
looks like the implementation is indeed incorrect (or doesn't match the documentation): https://github.com/openvswitch/ovs/blob/master/lib/meta-flow.c#L2331. Non trivial masks will always be rejected for the conntrack fields, except for the state, mark and label.

Reply:
Thanks for reporting this @wenyingd. It seems that it only affects parsing of flows in ovs-ofctl, which means you are able to support masks / CIDRs with libOpenflow?

Reply:
I have confirmed with OVS folk, using CIDR in such fields is not supported in the OVS implementations for OpenFlow, not only the ovs-ofctl command. We could use IP first, and add other APIs for IPNet params after OVS has fix on it.

Reply:
Ok, good thing we do not need it for now :)